src/printer.rs

```diff
@@ -892,14 +897,130 @@ impl Colors {
             ..Style::default()
         };
 
+        // Get scope style from theme using given `scope_name` or use `default_style`.
+        let get_scope_style = |scope_name: &str, default_style: Style| {
+            let scope_selectors = ScopeSelectors::from_str(scope_name).unwrap();
+
+            theme
+                .scopes
+                .iter()
+                .find(|item| item.scope.eq(&scope_selectors))
```

Comment:
Scope selectors from the theme generally shouldn't be expected to *exactly* match a scope, but instead use scope selector matching logic to determine which theme rules match and which has the highest precedence... Probably it doesn't matter too much here, but I would prefer consistency with how highlighting works

Reply:
It seems like I missed this reply earlier, sorry about that. I meant that the tmTheme may choose to have rules with less specific scope selectors like `markup` which would match `markup.deleted`, `markup.changed` etc. or even multiple rules could theoretically match, and whose styles would get applied/merged together etc. based on precedence order. So my idea was to [use proper scope selector logic](https://github.com/trishume/syntect/blob/master/src/highlighting/selector.rs#L34) instead of checking for string equality, but probably it isn't going to be necessary in practice, because, like you pointed out, most themes have these specific rules.

Reply:
I would prefer it, yes please 👍